### PR TITLE
try to cancel exisiting running or queued builds for a specific PR

### PR DESCRIPTION
**Pull request by jfrazelle**
try to cancel exisiting running or queued builds for a specific PR before rebuilding

ping @calavera @tiborvass

### CODE REVIEW

**Comment by jfrazelle** [comment]
jenkins api is gross af

**Comment by calavera** [comment]
:+1: :clap:

File: jenkins/jenkins.go [hunk 1]
```diff
@@ -112,3 +149,167 @@ func (c *Client) BuildWithParameters(job string, parameters string) error {
...
 }
+
+// CancelBuildsForPR cancels any queued or running builds for a PR.
+func (c *Client) CancelBuildsForPR(job, pr string) error {
+	var e string
+
+	// first check the queue
+	q, err := c.GetQueuedBuildForPR(job, pr)
+	if err != nil {
+		e += fmt.Sprintf("Getting queued build for job %s, pr %s failed: %v;", job, pr, err)
```
Comment (tiborvass) [inline jenkins/jenkins.go]: small nit but this could just be `e =` same for the `else if` block

Comment (tiborvass) [inline jenkins/jenkins.go]: also wonder if there isn't a `\n` missing, since you append to it further down.

Comment (jfrazelle) [inline jenkins/jenkins.go]: yeah but logrus does new lines bad :(

**Comment by jfrazelle** [comment]
dont merge im just going to use this branch to build new images and test

**Comment by tiborvass** [comment]
@jfrazelle was just wondering, when you force push a PR, it doesn't automatically go to the same server does it? So the CancelJob will not cancel job a that was started on another host right?

**Comment by jfrazelle** [comment]
The nodes don't matter they are just slaves the master will cancel it
regardless of where it is running

On Saturday, February 6, 2016, Tibor Vass <notifications@github.com> wrote:

> @jfrazelle <https://github.com/jfrazelle> was just wondering, when you
> force push a PR, it doesn't automatically go to the same server does it? So
> the CancelJob will not cancel job a that was started on another host right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/leeroy/pull/47#issuecomment-180909038>.
>


-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>

**Comment by calavera** [comment]
LGTM

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**